testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java

```diff
@@ -2183,6 +2183,85 @@ public void testTrinoAnalyze()
         onTrino().executeQuery("DROP TABLE " + trinoTableName);
     }
 
+    @Test(dataProvider = "storageFormats")
+    public void testRegisterTableSuccess(StorageFormat storageFormat)
+    {
+        String baseOldTableName = "test_register_table_success_old_" + randomTableSuffix() + "_" + storageFormat.name().toLowerCase(ENGLISH);
+        String trinoOldTableName = trinoTableName(baseOldTableName);
+        String sparkOldTableName = sparkTableName(baseOldTableName);
+
+        onTrino().executeQuery(format("CREATE TABLE %s (a int, b varchar, c boolean) with (format = '%s', partitioning = ARRAY['a'])", trinoOldTableName, storageFormat));
+        onTrino().executeQuery(format("INSERT INTO %s values(1, 'INDIA', true)", trinoOldTableName));
+        onTrino().executeQuery(format("INSERT INTO %s values(2, 'USA', false)", trinoOldTableName));
+        onTrino().executeQuery(format("COMMENT ON TABLE %s IS '%s'", trinoOldTableName, "This is table comment"));
+        onTrino().executeQuery(format("COMMENT ON COLUMN %s.%s IS '%s'", trinoOldTableName, "b", "This is column comment"));
+
+        List<Row> expected = List.of(row(1, "INDIA", true), row(2, "USA", false));
+
+        assertThat(onTrino().executeQuery(format("SELECT * FROM %s", trinoOldTableName)))
+                .containsOnly(expected);
+        assertThat(onSpark().executeQuery(format("SELECT * FROM %s", sparkOldTableName)))
+                .containsOnly(expected);
+
+        String tableLocation = getTableLocation(trinoOldTableName, true);
+
+        String baseNewTableName = "test_register_table_success_new_" + randomTableSuffix() + "_" + storageFormat.name().toLowerCase(ENGLISH);
```

Comment:
add the storage before the random table suffix

Reply:
Addressed!